### DISCUSSION TITLE
Choose custom download/maps folder

### DISCUSSION THREAD

**Discussion by maxbrito500** [organicmaps/organicmaps#8074] [Ideas]
Hello,

Organic Maps already permits to choose the memory card as external storage of downloaded maps. However, it does not allow the choose a custom folder inside the external memory card.

Why does this matter?

In communities where internet is either expensive to use or not even available, it is common practice to share the data physically through disks or through Wi-Fi on local networks.

Typically we use a folder called "box" where all sort of offline content can be found (e.g. books, movies, apps,..) and for an off-grid community the offline maps are critical.

Since right now the folder where maps is not possible to choose, it is very difficult for the common users to know in which folder to place the maps and reuse them without need for downloads. If we can customize the maps folder location, we can keep everything simple under the same shared folder.

I've looked on the beta releases, I've looked here on the forum but didn't found a similar request for a custom download location folder.

If you consider this request for implementation, we'd be very grateful, as well as any other off-grid community out there.

Thank you for the consideration and replies.

**Comment by biodranik** [comment]
Since Android 11 there are very strict system restrictions for apps to access any folders. AFAIR, there is no easy way to implement it.

**Comment by maxbrito500** [comment]
Thank you for the fast feedback. Oher offline apps like Tachiyomi adjusted to more recent androids. Perhaps looking on their code base can provide a good example?

Main difference seems to be requesting explicit permission to the user: https://stackoverflow.com/questions/72948052/android-13-read-external-storage-permission-still-usable

If

**Reply by biodranik** [reply]
That's not that easy. We won't add any read/write external storage permissions into the manifest due to security and privacy. Any apps that have these permissions can access any files on older androids. And steal any data.

**Comment by maxbrito500** [comment]
It is possible to request the storage permissions only when the user explicitly wishes to choose a custom folder.

This way doesn't affect the current usage model, and those who need a custom folder can then grant explicitly that permission.

Perhaps it would help to add a note or popup box with more info for the user before granting permission?

**Reply by biodranik** [reply]
1. Are there any open-source apps that do that properly without external storage permissions in the manifest?
2. Does this permission expire?

**Reply by maxbrito500** [reply]
1.Tachiyomi might be a good example: https://github.com/jobobby04/TachiyomiSY

I'm not sure about storage permissions in the manifest, please check. On android 13 I don't see storage permission requests when installing or using the app. This permission request just happens when choosing the custom folder.

Please try for yourself as my words might not be accurate enough to explain this.

2. Not that I know of. I think it is also scoped to the selected folder.

**Reply by biodranik** [reply]
This application has write external storage permission: https://github.com/jobobby04/TachiyomiSY/blob/master/app/src/main/AndroidManifest.xml#L12

Can you please remove this permission, together with other two mentions here: https://github.com/jobobby04/TachiyomiSY/blob/master/app/src/main/AndroidManifest.xml#L52
build the app, and tell us if it works? What is your android version?

**Reply by maxbrito500** [reply]
Thank you for the review.

I'm not an Android developer but will do as request and do the experiments. Please permit some time for getting this done but I'll reply back soon.

Thanks.
